segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessorTest.java

```diff
@@ -1506,6 +1521,128 @@ public void testReadTableEntriesDelta() throws Exception {
+        assertEquals(segment, created.getSegment());
+
+        UUID uuid = UUID.randomUUID();
+        WireCommands.AppendSetup setup = (WireCommands.AppendSetup) sendRequest(channel, new WireCommands.SetupAppend(2, uuid, segment, ""));
+        assertEquals(segment, setup.getSegment());
+        assertEquals(indexSegment, store.getStreamSegmentInfo(indexSegment, PravegaRequestProcessor.TIMEOUT).join().getName());
+        data.retain();
+
+        //Append 40 bytes of data
+        sendRequest(channel, new Append(segment, uuid, 1, new WireCommands.Event(data), 1L));
+        sendRequest(channel, new Append(segment, uuid, 2, new WireCommands.Event(data), 1L));
+
+        final long truncateOffset = -5000;
+        AssertExtensions.assertLessThan("Negative offset", 0, truncateOffset);
+
+        IllegalStateException exception = Assert.assertThrows(IllegalStateException.class, () ->
+                sendRequest(channel, new WireCommands.TruncateSegment(requestId, segment, truncateOffset, "")));
+        assertTrue(exception.getMessage().contains("No reply to request"));
+    }
+
+    @Test
+    public void testTruncateIndexSegmentWithGreaterThenEndOffset() throws Exception {
+        String segment = "testTruncateIndexSegmentWithGreaterThenEndOffset";
+        String indexSegment = getIndexSegmentName(segment);
+        ByteBuf data = Unpooled.wrappedBuffer("Hello world\n".getBytes());
+        ServiceBuilder serviceBuilder = newInlineExecutionInMemoryBuilder(getBuilderConfig());
+
+        serviceBuilder.initialize();
+        StreamSegmentStore store = serviceBuilder.createStreamSegmentService();
+        @Cleanup
+        EmbeddedChannel channel = createChannel(store);
+        WireCommands.SegmentCreated created = (WireCommands.SegmentCreated) sendRequest(channel, new WireCommands.CreateSegment(1, segment, WireCommands.CreateSegment.NO_SCALE, 0, "", 1024L));
+        assertEquals(segment, created.getSegment());
+
+        UUID uuid = UUID.randomUUID();
+        WireCommands.AppendSetup setup = (WireCommands.AppendSetup) sendRequest(channel, new WireCommands.SetupAppend(2, uuid, segment, ""));
+        assertEquals(segment, setup.getSegment());
+        assertEquals(indexSegment, store.getStreamSegmentInfo(indexSegment, PravegaRequestProcessor.TIMEOUT).join().getName());
+        data.retain();
+
+        //Append 40 bytes of data
+        sendRequest(channel, new Append(segment, uuid, 1, new WireCommands.Event(data), 1L));
+        sendRequest(channel, new Append(segment, uuid, 2, new WireCommands.Event(data), 1L));
+
+        final long truncateOffset = 100;
+        AssertExtensions.assertGreaterThan("Larger then endOffset", store.getStreamSegmentInfo(segment, PravegaRequestProcessor.TIMEOUT).join().getLength(), truncateOffset);
+        sendRequest(channel, new WireCommands.TruncateSegment(requestId, segment, truncateOffset, ""));
```

Comment:
please assert for exception, since the requested offset is larger.

Reply:
taken care.

Reply:
removed